app/src/main/java/com/example/pokemon/ui/MainActivity.kt

```diff
@@ -35,16 +31,16 @@ class MainActivity : AppCompatActivity(), MainContract.MainView {
         mainPresenterImpl = MainPresenterImpl(this)
         pokemonAdapter = PokemonAdapter(pokeList) {
             val intent = Intent(this@MainActivity, PokemonDetailActivity::class.java)
-            intent.putExtra("Pokemon", mainPresenterImpl.getPokemonList(it))
+            intent.putExtra("Pokemon", mainPresenterImpl!!.getPokemonList(it))
```

Comment:
don't use !! use a let check on mPresenter?.getPokemonList(it)?.let and then only proceed